packages/block-editor/src/hooks/utils.js

```diff
@@ -133,23 +133,56 @@ export function shouldSkipSerialization(
 	return skipSerialization;
 }
 
+const pendingStyleOverrides = new WeakMap();
+
 export function useStyleOverride( { id, css, assets, __unstableType } = {} ) {
 	const { setStyleOverride, deleteStyleOverride } = unlock(
 		useDispatch( blockEditorStore )
 	);
+	const registry = useRegistry();
 	const fallbackId = useId();
 	useEffect( () => {
 		// Unmount if there is CSS and assets are empty.
 		if ( ! css && ! assets ) return;
+
 		const _id = id || fallbackId;
-		setStyleOverride( _id, {
+		const override = {
 			id,
 			css,
 			assets,
 			__unstableType,
+		};
+		// Batch updates to style overrides to avoid triggering cascading renders
+		// for each style override block included in a tree and optimize initial render.
+		if ( ! pendingStyleOverrides.get( registry ) ) {
+			pendingStyleOverrides.set( registry, [] );
+		}
+		pendingStyleOverrides.get( registry ).push( [ _id, override ] );
+		window.queueMicrotask( () => {
```

Comment:
Why don't we micro task and batch all store updates generally?

Reply:
I think I've tried this in the past, I believe it results in some bugs in some situations where we expect the update to be synchronous, worth a try again though.